Add a11y test coverage to Rule Creation Flow for Detections tab

ecezalp opened the pull request:
This change adds a11y test coverage via FTR to the rule creation flow.

Relates to #80060.

To run the tests:
cd to x-pack
comment out lines 19 to 36 in `x-pack/test/accessibility/config.ts` to run only the security_solution suite
start the test server with `node scripts/functional_tests_server.js`
start the test with `node scripts/functional_test_runner --config test/accessibility/config.ts`

Additional Notes:

This test is a part of the Detections checkbox in the parent ticket #80060. However, you will notice that the checkbox is not actually checked, as the inclusion of this test does not cover the entire contents of the Detections tab, just the rule creation flow. In the future, a new ticket / PR should be created to ensure that mock alerts are loaded via esArchiver and the Detections tab should be further covered.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112638/)
* Commit: 414351831deb8415cf40552e6315534762c94f93
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112638/execution/node/430/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112638/","number":"112638","commit":"414351831deb8415cf40552e6315534762c94f93"}]}
PIPELINE-->

bhavyarm reviewed (commented):
@ecezalp we run the actual tests in test files and page objects are used as utilities. So this line ```  await this.a11y.testAppSnapshot(); ``` will be in security_solution.ts and not in various page objects.

ecezalp commented:
notes from our call with Bhavya:

- assertions should be moved from test objects into the body of the test
- one assertion should not be exceeded per `it` block
- core utilities on the DetectionPage object should be moved to appropriate places (such as find, testSubjects, common page)
- test scope should be limited to the major features: all modals, popovers, dropdowns on the page do not need to be clicked as there is extensive EUI a11y coverage that takes care of those
- rules do not need to be created, all that matters is the snapshot of the empty form
- [axe DevTools](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd) is a must-have tool to observe the page locally from an a11y standpoint
- if the Detections Page is not entirely covered by the ticket, a new ticket should be created and linked for the rest of the Detections page functionality
- data should be loaded via esArchiver tool for the Detections Overview page
- timeline should be covered

myasonik commented:
That's a great list above - thanks for meeting you two! :elastic-heart:

> test scope should be limited to the major features: all modals, popovers, dropdowns on the page do not need to be clicked as there is extensive EUI a11y coverage that takes care of those

I just want to clarify on this point (apologies if it was covered in your call) that though EUI coverage is taken care of, what you put into those EUI components isn't covered.

So, for example, you don't need to check that EUI's modal is accessible - that part is on EUI to ensure. But if you have a lot of content (particularly, interactive content) within that modal, you might still want to test it. (You'll have to use your judgement on the specifics here.)

And that's roughly the case with a lot of EUI's components. You don't need to double check their work (though you sometimes will just by nature of it being what our Solutions are built on) but you should still check your work as it all fits together.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113215/)
* Commit: 414351831deb8415cf40552e6315534762c94f93
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113215/execution/node/448/log)

### History
* :broken_heart: [Build #112638](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112638/) failed 414351831deb8415cf40552e6315534762c94f93

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112638/","number":"112638","commit":"414351831deb8415cf40552e6315534762c94f93"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113215/","number":"113215","commit":"414351831deb8415cf40552e6315534762c94f93"}]}
PIPELINE-->

bhavyarm commented:
@myasonik thanks for clarifying it. I was confused about KQL search bar, filter panel etc which we have already covered in other a11y tests.  I gotcha. Cheers!

elasticmachine commented:
Pinging @elastic/kibana-accessibility (Project:Accessibility)

elasticmachine commented:
Pinging @elastic/siem (Team:SIEM)

azasypkin commented on x-pack/test/security_functional/page_objects/detections/index.ts:
```diff
@@ -0,0 +1,157 @@
+/*
```
__issue:__ looks like this file should be placed somewhere in `x-pack/test/security_solution_*` folders, `x-pack/test/security_functional` is dedicated to Kibana Security Platform tests (`x-pack/plugins/security`).

ecezalp replied:
thank you for letting me know! file is moved to `x-pack/test/security_solution_ftr`

dmlemeshko commented on x-pack/test/security_solution_ftr/page_objects/detections/index.ts:
```diff
@@ -0,0 +1,157 @@
+    }
+
+    async selectMachineLearningJob(): Promise<void> {
+      const jobDropdown = await find.byCssSelector('[data-test-subj="mlJobSelect"] button');
+      await jobDropdown.click();
+      const firstJob = await find.byCssSelector('#high_distinct_count_error_message');
+      await firstJob.click();
```
```suggestion
      await find.clickByCssSelector('[data-test-subj="mlJobSelect"] button');
      await find.clickByCssSelector('#high_distinct_count_error_message');
```

dmlemeshko commented on x-pack/test/security_solution_ftr/page_objects/detections/index.ts:
```diff
@@ -0,0 +1,157 @@
+    }
+
+    async closeAddFilterPopover(): Promise<void> {
+      const closeButton = await testSubjects.find('cancelSaveFilter');
+      await closeButton.click();
```
```suggestion
      await testSubjects.click('cancelSaveFilter');
```

dmlemeshko commented on x-pack/test/security_solution_ftr/page_objects/detections/index.ts:
```diff
@@ -0,0 +1,157 @@
+    async replaceIndexPattern(): Promise<void> {
+      const buttons = await find.allByCssSelector('[data-test-subj="comboBoxInput"] button');
+      await buttons.map(async (button: WebElementWrapper) => await button.click());
+      const input = await find.byCssSelector('input[data-test-subj="comboBoxSearchInput"]');
+      await input.type('*');
```
```suggestion
      await testSubjects.setValue('comboBoxSearchInput', '*');
```

dmlemeshko commented on x-pack/test/security_solution_ftr/page_objects/detections/index.ts:
```diff
@@ -0,0 +1,157 @@
+    }
+
+    async toggleSavedQueries(): Promise<void> {
+      const filterActions = await find.allByCssSelector(
+        '[data-test-subj="saved-query-management-popover-button"]'
+      );
+      await filterActions[1].click();
```
```suggestion
      const filterActions = await testSubjects.findAll('saved-query-management-popover-button"');
      await filterActions[1].click();
```

dmlemeshko replied:
Why do we click specifically the 2nd element?

ecezalp replied:
because there seems to be three of these buttons, and clicking the second one opens the popover

dmlemeshko replied:
Got it. Could we set unique data-test-subj based on the action?

dmlemeshko commented on x-pack/test/security_solution_ftr/page_objects/detections/index.ts:
```diff
@@ -0,0 +1,157 @@
+    }
+
+    async addCustomQuery(query: string): Promise<void> {
+      const textAreaElement = await testSubjects.find('queryInput');
+      await textAreaElement.type(query);
```
```suggestion
      await testSubjects.setValue('queryInput', query);
```

dmlemeshko commented on x-pack/test/security_solution_ftr/page_objects/detections/index.ts:
```diff
@@ -0,0 +1,157 @@
+      name: string = 'test rule name',
+      description: string = 'test rule description'
+    ): Promise<void> {
+      const nameInput = await find.byCssSelector(
+        `[aria-describedby="detectionEngineStepAboutRuleName"]`
+      );
+      await nameInput.type(name);
+      const descriptionTextArea = await find.byCssSelector(
+        `[aria-describedby="detectionEngineStepAboutRuleDescription"]`
+      );
+      await descriptionTextArea.type(description);
```
```suggestion
      await find.setValue(`[aria-describedby="detectionEngineStepAboutRuleName"]`, name);
            await find.byCssSelector(
        `[aria-describedby="detectionEngineStepAboutRuleDescription"]`,
        description
      );
```

dmlemeshko commented on x-pack/test/security_solution_ftr/page_objects/detections/index.ts:
```diff
@@ -0,0 +1,157 @@
+    }
+
+    async addEqlQuery(query: string): Promise<void> {
+      const textArea = await find.byCssSelector(
+        '[data-test-subj="detectionEngineStepDefineRuleEqlQueryBar"] [data-test-subj="eqlQueryBarTextInput"]'
+      );
+      await textArea.type(query);
```
```suggestion
      const textArea = await testSubjects.setValue(
        'detectionEngineStepDefineRuleEqlQueryBar > eqlQueryBarTextInput',
        query
      );
```

dmlemeshko replied:
More examples here :)

https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/packages/kbn-test-subj-selector/index.test.js#L13-L22

dmlemeshko commented on x-pack/test/security_solution_ftr/page_objects/detections/index.ts:
```diff
@@ -0,0 +1,157 @@
+    }
+
+    async openAddFilterPopover(): Promise<void> {
+      const addButtons = await find.allByCssSelector('[data-test-subj="addFilter"]');
```
```suggestion
      const addButtons = await testSubjects.findAll('addFilter');
```

dmlemeshko commented on x-pack/test/security_solution_ftr/page_objects/detections/index.ts:
```diff
@@ -0,0 +1,157 @@
+    }
+
+    async toggleFilterActions(): Promise<void> {
+      const filterActions = await find.allByCssSelector('[data-test-subj="addFilter"]');
```
```suggestion
      const filterActions = await testSubjects.findAll('addFilter');
```

dmlemeshko reviewed (changes requested):
Thank you for adding tests!

I added minor suggestions to make code smaller and use `testSubjects` over `find` whenever possible

ecezalp commented:
hey @dmlemeshko, thank you so much for your suggestions, I have incorporated all suggestions, and I was very happy about how the methods came out. However I seem to run into a quite strange issue. I rebased against master and I am now unable to run the tests.

This is what happens on the first assertion
<img width="1032" alt="Screen Shot 2021-03-17 at 2 44 51 PM" src="https://user-images.githubusercontent.com/18617331/111521572-aa947200-872f-11eb-8b95-0edcde0a6445.png">

and this is what happens if I navigate manually on the test browser to '/'
<img width="1030" alt="Screen Shot 2021-03-17 at 2 45 06 PM" src="https://user-images.githubusercontent.com/18617331/111521642-bbdd7e80-872f-11eb-8044-4d46bc69166e.png">

where .. did the security solution go? is this expected?? Please let me know!

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113702/)
* Commit: 20b416b971dbda3847ad0545431a30d998968858
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94377/20b416b971dbda3847ad0545431a30d998968858)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-accessibility](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113702/execution/node/678/log)
* [Execute kibana-ciGroup10](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113702/execution/node/628/log)

### History
* :broken_heart: [Build #113215](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113215/) failed 414351831deb8415cf40552e6315534762c94f93
* :broken_heart: [Build #112638](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112638/) failed 414351831deb8415cf40552e6315534762c94f93

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @ecezalp

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112638/","number":"112638","commit":"414351831deb8415cf40552e6315534762c94f93"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113215/","number":"113215","commit":"414351831deb8415cf40552e6315534762c94f93"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113702/","number":"113702","commit":"20b416b971dbda3847ad0545431a30d998968858"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113773/)
* Commit: 952a07f6eda52ea94d4cd7d491a2871276127b9d
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94377/952a07f6eda52ea94d4cd7d491a2871276127b9d)
* This comment will update when the build is complete

### Failed CI Steps
* [fetch latest from 'master' at origin](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113773/execution/node/130/log)
* [Execute xpack-accessibility](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113773/execution/node/679/log)
* [Execute xpack-accessibility](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113773/execution/node/827/log)

### History
* :broken_heart: [Build #113702](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113702/) failed 20b416b971dbda3847ad0545431a30d998968858
* :broken_heart: [Build #113215](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113215/) failed 414351831deb8415cf40552e6315534762c94f93
* :broken_heart: [Build #112638](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112638/) failed 414351831deb8415cf40552e6315534762c94f93

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @ecezalp

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112638/","number":"112638","commit":"414351831deb8415cf40552e6315534762c94f93"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113215/","number":"113215","commit":"414351831deb8415cf40552e6315534762c94f93"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113702/","number":"113702","commit":"20b416b971dbda3847ad0545431a30d998968858"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113773/","number":"113773","commit":"952a07f6eda52ea94d4cd7d491a2871276127b9d"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113714/)
* Commit: 2e50c521a8ba33b7343da5a65018bf701ca0b6b7
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94377/2e50c521a8ba33b7343da5a65018bf701ca0b6b7)
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113714/execution/node/445/log)

### History
* :broken_heart: [Build #113773](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113773/) failed 952a07f6eda52ea94d4cd7d491a2871276127b9d
* :broken_heart: [Build #113702](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113702/) failed 20b416b971dbda3847ad0545431a30d998968858
* :broken_heart: [Build #113215](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113215/) failed 414351831deb8415cf40552e6315534762c94f93
* :broken_heart: [Build #112638](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112638/) failed 414351831deb8415cf40552e6315534762c94f93

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @ecezalp

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112638/","number":"112638","commit":"414351831deb8415cf40552e6315534762c94f93"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113215/","number":"113215","commit":"414351831deb8415cf40552e6315534762c94f93"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113702/","number":"113702","commit":"20b416b971dbda3847ad0545431a30d998968858"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113773/","number":"113773","commit":"952a07f6eda52ea94d4cd7d491a2871276127b9d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113714/","number":"113714","commit":"2e50c521a8ba33b7343da5a65018bf701ca0b6b7"}]}
PIPELINE-->

dmlemeshko commented on x-pack/test/security_solution_ftr/page_objects/detections/index.ts:
```diff
@@ -0,0 +1,149 @@
+      description: string = 'test rule description'
+    ): Promise<void> {
+      await find.setValue(`[aria-describedby="detectionEngineStepAboutRuleName"]`, name);
+      const input = await find.byCssSelector(
+        `[aria-describedby="detectionEngineStepAboutRuleDescription"]`
+      );
+      await input.type(description);
```
nit:
```suggestion
      await find.setValue(
        `[aria-describedby="detectionEngineStepAboutRuleDescription"]`,
        description
      );
```

ecezalp replied:
added!

dmlemeshko reviewed (commented):
Code looks good. 

While checking [CI logs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113773/execution/node/679/log/?consoleFull), I found this error on start:

```
20:18:10               │ proc [kibana]   log   [19:18:10.420] [info][savedobjects-service] [.kibana] MARK_VERSION_INDEX_READY -> DONE
20:18:10               │ proc [kibana]   log   [19:18:10.421] [info][savedobjects-service] [.kibana] Migration completed after 943ms
20:18:13               │ proc [kibana]   log   [19:18:12.916] [error][data][data][indexPatterns][plugins] ResponseError: security_exception
20:18:13               │ proc [kibana]     at onBody (/dev/shm/workspace/kibana-build-xpack-4/node_modules/@elastic/elasticsearch/lib/Transport.js:333:23)
20:18:13               │ proc [kibana]     at IncomingMessage.onEnd (/dev/shm/workspace/kibana-build-xpack-4/node_modules/@elastic/elasticsearch/lib/Transport.js:260:11)
20:18:13               │ proc [kibana]     at IncomingMessage.emit (events.js:327:22)
20:18:13               │ proc [kibana]     at endReadableNT (internal/streams/readable.js:1327:12)
20:18:13               │ proc [kibana]     at processTicksAndRejections (internal/process/task_queues.js:80:21) {
20:18:13               │ proc [kibana]   meta: {
20:18:13               │ proc [kibana]     body: { error: [Object], status: 401 },
20:18:13               │ proc [kibana]     statusCode: 401,
20:18:13               │ proc [kibana]     headers: {
20:18:13               │ proc [kibana]       'x-opaque-id': '2aee5ba3-aeff-4cd4-ba3c-8ac367186bd8',
20:18:13               │ proc [kibana]       'www-authenticate': 'Basic realm="security" charset="UTF-8", ApiKey',
20:18:13               │ proc [kibana]       'content-type': 'application/json;charset=utf-8',
20:18:13               │ proc [kibana]       'content-length': '463'
20:18:13               │ proc [kibana]     },
20:18:13               │ proc [kibana]     meta: {
20:18:13               │ proc [kibana]       context: null,
20:18:13               │ proc [kibana]       request: [Object],
20:18:13               │ proc [kibana]       name: 'elasticsearch-js',
20:18:13               │ proc [kibana]       connection: [Object],
20:18:13               │ proc [kibana]       attempts: 0,
20:18:13               │ proc [kibana]       aborted: false
20:18:13               │ proc [kibana]     }
20:18:13               │ proc [kibana]   },
20:18:13               │ proc [kibana]   isBoom: true,
20:18:13               │ proc [kibana]   isServer: true,
20:18:13               │ proc [kibana]   data: null,
20:18:13               │ proc [kibana]   output: {
20:18:13               │ proc [kibana]     statusCode: 500,
20:18:13               │ proc [kibana]     payload: {
20:18:13               │ proc [kibana]       statusCode: 500,
20:18:13               │ proc [kibana]       error: 'Internal Server Error',
20:18:13               │ proc [kibana]       message: 'An internal server error occurred'
20:18:13               │ proc [kibana]     },
20:18:13               │ proc [kibana]     headers: {}
20:18:13               │ proc [kibana]   },
20:18:13               │ proc [kibana]   [Symbol(SavedObjectsClientErrorCode)]: 'SavedObjectsClient/generalError'
20:18:13               │ proc [kibana] }
```

It does not fail the tests, but what does it mean? Can we change FTR config to avoid it?

ecezalp commented:
@dmlemeshko yes I became aware of that error recently, and just pushed a fix. I had removed the line `common.navigateToApp('security')` thinking that I could just navigate to the url anyway, but that turned out to be a mistake as I believe removing that line disabled the security_solutions plugin, and broke the test quite badly! It should be good now.

Added the last suggestion as well. Thank you so much for your thorough review <3

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113998/)
* Commit: 16889b57fc08ca0ad9de061a1d198263b6430e5e
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94377/16889b57fc08ca0ad9de061a1d198263b6430e5e)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-accessibility](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113998/execution/node/675/log)
* [Execute xpack-firefox](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113998/execution/node/679/log)

### History
* :broken_heart: [Build #113773](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113773/) failed 952a07f6eda52ea94d4cd7d491a2871276127b9d
* :broken_heart: [Build #113714](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113714/) failed 2e50c521a8ba33b7343da5a65018bf701ca0b6b7
* :broken_heart: [Build #113702](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113702/) failed 20b416b971dbda3847ad0545431a30d998968858
* :broken_heart: [Build #113215](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113215/) failed 414351831deb8415cf40552e6315534762c94f93
* :broken_heart: [Build #112638](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112638/) failed 414351831deb8415cf40552e6315534762c94f93

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @ecezalp

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112638/","number":"112638","commit":"414351831deb8415cf40552e6315534762c94f93"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113215/","number":"113215","commit":"414351831deb8415cf40552e6315534762c94f93"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113702/","number":"113702","commit":"20b416b971dbda3847ad0545431a30d998968858"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113714/","number":"113714","commit":"2e50c521a8ba33b7343da5a65018bf701ca0b6b7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113773/","number":"113773","commit":"952a07f6eda52ea94d4cd7d491a2871276127b9d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113998/","number":"113998","commit":"16889b57fc08ca0ad9de061a1d198263b6430e5e"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114071/)
* Commit: 4c0d136e7927148b7ae356c9c37c54f09b47c261
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94377/4c0d136e7927148b7ae356c9c37c54f09b47c261)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-firefox](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114071/execution/node/682/log)
* [Execute xpack-firefox](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114071/execution/node/814/log)

### History
* :broken_heart: [Build #113998](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113998/) failed 16889b57fc08ca0ad9de061a1d198263b6430e5e
* :broken_heart: [Build #113773](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113773/) failed 952a07f6eda52ea94d4cd7d491a2871276127b9d
* :broken_heart: [Build #113714](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113714/) failed 2e50c521a8ba33b7343da5a65018bf701ca0b6b7
* :broken_heart: [Build #113702](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113702/) failed 20b416b971dbda3847ad0545431a30d998968858
* :broken_heart: [Build #113215](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113215/) failed 414351831deb8415cf40552e6315534762c94f93

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @ecezalp

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113215/","number":"113215","commit":"414351831deb8415cf40552e6315534762c94f93"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113702/","number":"113702","commit":"20b416b971dbda3847ad0545431a30d998968858"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113714/","number":"113714","commit":"2e50c521a8ba33b7343da5a65018bf701ca0b6b7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113773/","number":"113773","commit":"952a07f6eda52ea94d4cd7d491a2871276127b9d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113998/","number":"113998","commit":"16889b57fc08ca0ad9de061a1d198263b6430e5e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114071/","number":"114071","commit":"4c0d136e7927148b7ae356c9c37c54f09b47c261"}]}
PIPELINE-->

ecezalp commented:
Esteemed Reviewers,

Please note that I squashed some of the changes suggested - which could impact your review flow. My apologies! Please feel free to review this ticket as is for now, and any further changes that may come from any further suggestions will no longer be squashed for your reviewing pleasure 😊

ecezalp commented:
@elasticmachine merge upstream

rylnd commented:
One quick note about review steps: both the test server and runner need to be provided the same config:

```bash
cd x-pack
node scripts/functional_tests_server --config test/accessibility/config.ts
node scripts/functional_test_runner --config test/accessibility/config.ts
```

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114108/)
* Commit: 9e4266e5f43e88681d0f0653d6868b2d82b03de9
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94377/9e4266e5f43e88681d0f0653d6868b2d82b03de9)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #114071](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114071/) failed 4c0d136e7927148b7ae356c9c37c54f09b47c261
* :broken_heart: [Build #113998](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113998/) failed 16889b57fc08ca0ad9de061a1d198263b6430e5e
* :broken_heart: [Build #113773](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113773/) failed 952a07f6eda52ea94d4cd7d491a2871276127b9d
* :broken_heart: [Build #113714](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113714/) failed 2e50c521a8ba33b7343da5a65018bf701ca0b6b7
* :broken_heart: [Build #113702](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113702/) failed 20b416b971dbda3847ad0545431a30d998968858

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @ecezalp

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113702/","number":"113702","commit":"20b416b971dbda3847ad0545431a30d998968858"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113714/","number":"113714","commit":"2e50c521a8ba33b7343da5a65018bf701ca0b6b7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113773/","number":"113773","commit":"952a07f6eda52ea94d4cd7d491a2871276127b9d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113998/","number":"113998","commit":"16889b57fc08ca0ad9de061a1d198263b6430e5e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114071/","number":"114071","commit":"4c0d136e7927148b7ae356c9c37c54f09b47c261"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114108/","number":"114108","commit":"9e4266e5f43e88681d0f0653d6868b2d82b03de9"}]}
PIPELINE-->

rylnd commented:
In the course of testing locally, I found that I was getting frequent test failures, mostly related to elements not being clickable:

<img width="1252" alt="Default__-bash_" src="https://user-images.githubusercontent.com/657252/111707780-b147e580-8812-11eb-835c-ca1af470a87b.png">

and after some pairing with @ecezalp it appears that the issue is due to our fixed timeline footer at the bottom of the page:

![Mar-18-2021 17-50-43](https://user-images.githubusercontent.com/657252/111707685-865d9180-8812-11eb-97e2-aa09028d7af5.gif)

At this point I believe the pathology is:

1. Test requests that an element below the fold be clicked
2. Page driver navigates so that the element is within the viewport
3. A click on the element's bounding box is attempted, but the footer is occluding the element and intercepts the click

At this point I'm not sure if this is a bug with our CSS, or with the test/framework code, or whether the footer's occlusion represents a legitimate a11y issue.

@bhavyarm if you have any recommendations at this point, that would be appreciated!

ecezalp commented:
> At this point I'm not sure if this is a bug with our CSS, or with the test/framework code, or whether the footer's occlusion represents a legitimate a11y issue.

I believe that we can bypass the issue by adding an extra scroll of timeline bar height pixels when scrolling to any element that is below the viewport size minus timeline bar height. I don't think that this is a legitimate a11y issue, it's just that the test framework is not smart enough to deduce that a higher z-index fixed element actually constrains the viewport. The method can be placed on the DetectionsPage object. @bhavyarm @rylnd please let me know what you think and I would be happy to implement the fix.

MadameSheema commented on x-pack/test/accessibility/apps/security_solution.ts:
```diff
@@ -0,0 +1,140 @@
+        });
+
+        describe('Custom Query Rule', () => {
+          describe('First Step', () => {
```
For readability purposes I think that instead of `First Step`, `Second Step` and so on, it is better to use the name of the step: `Define step`, `About step` and so on.

myasonik reviewed (approved):
Just wanted to say these tests are looking awesome! Thanks to everyone involved with writing/reviewing/shipping these tests!

bhavyarm commented:
@elasticmachine merge upstream

bhavyarm commented:
So I am not sure what is happening. But couldn't get the test to run. It fails with Kibana didn't load properly and error message. 
```
   │ proc [kibana] server    log   [17:29:09.016] [error][data][data][indexPatterns][plugins] ResponseError: security_exception
   │ proc [kibana]     at onBody (/Users/bhavyarajumandya/Desktop/kibana_dev/kibana/node_modules/@elastic/elasticsearch/lib/Transport.js:333:23)
   │ proc [kibana]     at IncomingMessage.onEnd (/Users/bhavyarajumandya/Desktop/kibana_dev/kibana/node_modules/@elastic/elasticsearch/lib/Transport.js:260:11)
   │ proc [kibana]     at IncomingMessage.emit (events.js:327:22)
   │ proc [kibana]     at endReadableNT (internal/streams/readable.js:1327:12)
   │ proc [kibana]     at processTicksAndRejections (internal/process/task_queues.js:80:21) {
   │ proc [kibana]   meta: {
   │ proc [kibana]     body: { error: [Object], status: 401 },
   │ proc [kibana]     statusCode: 401,
   │ proc [kibana]     headers: {
   │ proc [kibana]       'x-opaque-id': '5fc06908-7fa4-4986-9975-6e0076feb94d',
   │ proc [kibana]       'www-authenticate': 'Basic realm="security" charset="UTF-8", ApiKey',
   │ proc [kibana]       'content-type': 'application/json;charset=utf-8',
   │ proc [kibana]       'content-length': '463'
   │ proc [kibana]     },
   │ proc [kibana]     meta: {
   │ proc [kibana]       context: null,
   │ proc [kibana]       request: [Object],
   │ proc [kibana]       name: 'elasticsearch-js',
   │ proc [kibana]       connection: [Object],
   │ proc [kibana]       attempts: 0,
   │ proc [kibana]       aborted: false
   │ proc [kibana]     }
   │ proc [kibana]   },
   │ proc [kibana]   isBoom: true,
   │ proc [kibana]   isServer: true,
   │ proc [kibana]   data: null,
   │ proc [kibana]   output: {
   │ proc [kibana]     statusCode: 500,
   │ proc [kibana]     payload: {
   │ proc [kibana]       statusCode: 500,
   │ proc [kibana]       error: 'Internal Server Error',
   │ proc [kibana]       message: 'An internal server error occurred'
   │ proc [kibana]     },
   │ proc [kibana]     headers: {}
   │ proc [kibana]   },
   │ proc [kibana]   [Symbol(SavedObjectsClientErrorCode)]: 'SavedObjectsClient/generalError'
```
I merged in main into ece's branch and bootstrapped again and same problem. @dmlemeshko do you know whats happening? @ecezalp will you keep me posted after you get a run in your local again? Thanks!

bhavyarm commented:
@ecezalp got the tests to run. You need to add this line in the before method.
```
await security.testUser.setRoles(['superuser'], false);
```

and in after
```
await security.testUser.restoreDefaults(false);
```

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114749/)
* Commit: 593bfc2a118c9906e34b6449db8a194deca21c1b
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94377/593bfc2a118c9906e34b6449db8a194deca21c1b)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #114108](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114108/) succeeded 9e4266e5f43e88681d0f0653d6868b2d82b03de9
* :broken_heart: [Build #114071](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114071/) failed 4c0d136e7927148b7ae356c9c37c54f09b47c261
* :broken_heart: [Build #113998](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113998/) failed 16889b57fc08ca0ad9de061a1d198263b6430e5e
* :broken_heart: [Build #113773](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113773/) failed 952a07f6eda52ea94d4cd7d491a2871276127b9d
* :broken_heart: [Build #113714](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113714/) failed 2e50c521a8ba33b7343da5a65018bf701ca0b6b7

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @ecezalp

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113714/","number":"113714","commit":"2e50c521a8ba33b7343da5a65018bf701ca0b6b7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113773/","number":"113773","commit":"952a07f6eda52ea94d4cd7d491a2871276127b9d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113998/","number":"113998","commit":"16889b57fc08ca0ad9de061a1d198263b6430e5e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114071/","number":"114071","commit":"4c0d136e7927148b7ae356c9c37c54f09b47c261"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114108/","number":"114108","commit":"9e4266e5f43e88681d0f0653d6868b2d82b03de9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114749/","number":"114749","commit":"593bfc2a118c9906e34b6449db8a194deca21c1b"}]}
PIPELINE-->

rylnd commented:
@bhavyarm are you still seeing sporadic local failures, or is that fixed in the latest?

bhavyarm commented:
@rylnd yes. Zoomed with Ece and @dmlemeshko. Ece will be adding some kind of scroll to find the element in viewport. The test will also be run in flaky test runner for the stability and needs to run on its own for atomicity. Thanks and cheers!

ecezalp commented:
notes from our meeting today with @bhavyarm and @dmlemeshko 
- we will use `common.scrollKibanaBodyTop` or a similar scroll method such as `browser.setScrollToById` to ensure that the tests can always run locally
- once tests are confirmed to be running fine locally, we will use the flaky test runner job to ensure that test is stable in CI as well
- once the above steps are confirmed, test will be ready to merge

ecezalp commented:
@elasticmachine merge upstream

ecezalp commented:
latest changes:

- was able to get to the bottom of the issue, it was the `click` method that scrolled any off-screen elements right under the navbar. It looks like there was an ability to add a top offset under the hood, but it wasn't exposed. so I was able to expose the ability to pass an offset to the scroll within the click method, and by giving a generous topOffset to any off-screen elements in the DetectionsPage, I believe I was able to permanently resolve the issue.
-  updated ambiguous rule step names with more explicit ones
- removed browser resizing in the before method as it is no longer necessary

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115064/)
* Commit: 3ccc4937689ea7fbe274dd4bb2a02b0bea57dd21
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115064/execution/node/441/log)

### History
* :green_heart: [Build #114749](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114749/) succeeded 593bfc2a118c9906e34b6449db8a194deca21c1b
* :green_heart: [Build #114108](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114108/) succeeded 9e4266e5f43e88681d0f0653d6868b2d82b03de9
* :broken_heart: [Build #114071](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114071/) failed 4c0d136e7927148b7ae356c9c37c54f09b47c261
* :broken_heart: [Build #113998](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113998/) failed 16889b57fc08ca0ad9de061a1d198263b6430e5e
* :broken_heart: [Build #113773](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113773/) failed 952a07f6eda52ea94d4cd7d491a2871276127b9d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @ecezalp

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113773/","number":"113773","commit":"952a07f6eda52ea94d4cd7d491a2871276127b9d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113998/","number":"113998","commit":"16889b57fc08ca0ad9de061a1d198263b6430e5e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114071/","number":"114071","commit":"4c0d136e7927148b7ae356c9c37c54f09b47c261"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114108/","number":"114108","commit":"9e4266e5f43e88681d0f0653d6868b2d82b03de9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114749/","number":"114749","commit":"593bfc2a118c9906e34b6449db8a194deca21c1b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115064/","number":"115064","commit":"3ccc4937689ea7fbe274dd4bb2a02b0bea57dd21"}]}
PIPELINE-->

bhavyarm commented:
@ecezalp https://kibana-ci.elastic.co/job/kibana+flaky-test-suite-runner/1395/ 
Finally a11y failure :)   cc @myasonik 

```
fail: Security Solution Detections Create Rule Flow Custom Query Rule Define Step default view meets a11y requirements
[00:12:37]                     │      Error: a11y report:
[00:12:37]                     │ 
[00:12:37]                     │ VIOLATION
[00:12:37]                     │   [aria-hidden-focus]: Ensures aria-hidden elements do not contain focusable elements
[00:12:37]                     │     Help: https://dequeuniversity.com/rules/axe/4.0/aria-hidden-focus?application=axeAPI
[00:12:37]                     │     Elements:
[00:12:37]                     │       - .hide-for-sharing
[00:12:37]                     │       - .Wrapper-o5t5gh-0
[00:12:37]                     │       - main > div[data-focus-guard="true"][data-aria-hidden="true"][aria-hidden="true"]:nth-child(1)
[00:12:37]                     │       - main > div[data-focus-guard="true"][data-aria-hidden="true"][aria-hidden="true"]:nth-child(2)
[00:12:37]                     │       - main > div[data-focus-guard="true"][data-aria-hidden="true"][aria-hidden="true"]:nth-child(4)
[00:12:37]                     │       - .Container-oo43db-0
[00:12:37]                     │       - .Wrapper-sc-5fa2oy-0
[00:12:37]                     │       at Accessibility.testAxeReport (/dev/shm/workspace/parallel/5/kibana/test/accessibility/services/a11y/a11y.ts:76:15)
[00:12:37]                     │       at Accessibility.testAppSnapshot (/dev/shm/workspace/parallel/5/kibana/test/accessibility/services/a11y/a11y.ts:49:18)
[00:12:37]                     │       at processTicksAndRejections (internal/process/task_queues.js:93:5)
[00:12:37]                     │       at Context.<anonymous> (test/accessibility/apps/security_solution.ts:34:15)
[00:12:37]                     │       at Object.apply (/dev/shm/workspace/parallel/5/kibana/node_modules/@kbn/test/src/functional_test_runner/lib/mocha/wrap_function.js:73:16)
[00:12:37]                     │ 
[00:12:37]                     │ 

```

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115072/)
* Commit: c776d09a0a93441448412a67404faa7b6acedc12
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94377/c776d09a0a93441448412a67404faa7b6acedc12)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #115064](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115064/) failed 3ccc4937689ea7fbe274dd4bb2a02b0bea57dd21
* :green_heart: [Build #114749](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114749/) succeeded 593bfc2a118c9906e34b6449db8a194deca21c1b
* :green_heart: [Build #114108](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114108/) succeeded 9e4266e5f43e88681d0f0653d6868b2d82b03de9
* :broken_heart: [Build #114071](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114071/) failed 4c0d136e7927148b7ae356c9c37c54f09b47c261
* :broken_heart: [Build #113998](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113998/) failed 16889b57fc08ca0ad9de061a1d198263b6430e5e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @ecezalp

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113998/","number":"113998","commit":"16889b57fc08ca0ad9de061a1d198263b6430e5e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114071/","number":"114071","commit":"4c0d136e7927148b7ae356c9c37c54f09b47c261"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114108/","number":"114108","commit":"9e4266e5f43e88681d0f0653d6868b2d82b03de9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114749/","number":"114749","commit":"593bfc2a118c9906e34b6449db8a194deca21c1b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115064/","number":"115064","commit":"3ccc4937689ea7fbe274dd4bb2a02b0bea57dd21"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115072/","number":"115072","commit":"c776d09a0a93441448412a67404faa7b6acedc12"}]}
PIPELINE-->

myasonik commented:
That looks like the bug we get from toasts.

Try bringing in the `toasts` service and calling `await toasts.dismissAllToasts();` right before this snapshot happens. 

If that doesn't work and you're not sure what else might be causing it, feel free to DM me 👍

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115205/)
* Commit: 3795373138dc92871963a2063244ea04c9470a92
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94377/3795373138dc92871963a2063244ea04c9470a92)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-accessibility](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115205/execution/node/697/log)
* [Execute xpack-accessibility](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115205/execution/node/828/log)

### History
* :green_heart: [Build #115072](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115072/) succeeded c776d09a0a93441448412a67404faa7b6acedc12
* :broken_heart: [Build #115064](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115064/) failed 3ccc4937689ea7fbe274dd4bb2a02b0bea57dd21
* :green_heart: [Build #114749](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114749/) succeeded 593bfc2a118c9906e34b6449db8a194deca21c1b
* :green_heart: [Build #114108](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114108/) succeeded 9e4266e5f43e88681d0f0653d6868b2d82b03de9
* :broken_heart: [Build #114071](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114071/) failed 4c0d136e7927148b7ae356c9c37c54f09b47c261

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @ecezalp

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114071/","number":"114071","commit":"4c0d136e7927148b7ae356c9c37c54f09b47c261"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114108/","number":"114108","commit":"9e4266e5f43e88681d0f0653d6868b2d82b03de9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114749/","number":"114749","commit":"593bfc2a118c9906e34b6449db8a194deca21c1b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115064/","number":"115064","commit":"3ccc4937689ea7fbe274dd4bb2a02b0bea57dd21"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115072/","number":"115072","commit":"c776d09a0a93441448412a67404faa7b6acedc12"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115205/","number":"115205","commit":"3795373138dc92871963a2063244ea04c9470a92"}]}
PIPELINE-->

ecezalp commented:
@elasticmachine merge upstream

bhavyarm reviewed (approved):
LGTM!!!! Woooooohoooo. Great work @ecezalp <3

dmlemeshko reviewed (approved):
LGTM

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115259/)
* Commit: d075281d695a7edca65550bd0b3f41b319cf9f1f
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94377/d075281d695a7edca65550bd0b3f41b319cf9f1f)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #115205](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115205/) failed 3795373138dc92871963a2063244ea04c9470a92
* :green_heart: [Build #115072](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115072/) succeeded c776d09a0a93441448412a67404faa7b6acedc12
* :broken_heart: [Build #115064](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115064/) failed 3ccc4937689ea7fbe274dd4bb2a02b0bea57dd21
* :green_heart: [Build #114749](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114749/) succeeded 593bfc2a118c9906e34b6449db8a194deca21c1b
* :green_heart: [Build #114108](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114108/) succeeded 9e4266e5f43e88681d0f0653d6868b2d82b03de9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @ecezalp

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114108/","number":"114108","commit":"9e4266e5f43e88681d0f0653d6868b2d82b03de9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114749/","number":"114749","commit":"593bfc2a118c9906e34b6449db8a194deca21c1b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115064/","number":"115064","commit":"3ccc4937689ea7fbe274dd4bb2a02b0bea57dd21"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115072/","number":"115072","commit":"c776d09a0a93441448412a67404faa7b6acedc12"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115205/","number":"115205","commit":"3795373138dc92871963a2063244ea04c9470a92"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115259/","number":"115259","commit":"d075281d695a7edca65550bd0b3f41b319cf9f1f"}]}
PIPELINE-->

kibanamachine commented:
Looks like this PR has a backport PR but it still hasn't been merged. Please merge it ASAP to keep the branches relatively in sync.

rylnd commented:
@ecezalp it looks like this didn't auto-merge due to the CI failures. Can you take a look?

rylnd commented:
@bhavyarm @myasonik @dmlemeshko this branch was failing consistently on the 7.x branch until an explicit browser resize was added. Might you have any insight into an explanation for that behavior? I just want to ensure that we understand the nature of the failures and that our browser resize fix is truly warranted/working.

elasticmachine commented:
Pinging @elastic/security-solution (Team: SecuritySolution)
